pytorch_lightning/loops/epoch/evaluation_epoch_loop.py

```diff
@@ -165,20 +163,16 @@ def on_save_checkpoint(self) -> Dict:
         state_dict = super().on_save_checkpoint()
 
         if (
-            self.trainer is None
-            or not self.trainer.state._fault_tolerant_mode.is_enabled
-            or self._data_fetcher is None
-            or self._num_completed_batches_reached()  # did not finish
-            # TODO: fault-tolerance requires a minimum number of batches so probably should be > 0
-            or self.batch_progress.current.ready == 0  # did not start
+            self.trainer is not None
```

Comment:
it is just making it inverse?

Reply:
No, it also got rid of

`        # TODO: this should use `pytorch_lightning/trainer/supporters.py::CombinedLoader._state_dict_fn``